### PR TITLE
Structural changes for Rest API

### PR DESCRIPTION
**Pull request by aaron-schnieder**
- Restructured /src into /game, updated dependency references and removed unnecessary files.
- Created /api and implemented placeholder Rest API server + method
- Removed Cargo.lock

### CODE REVIEW

**Comment by aaron-schnieder** [comment]
@DanMass @RichCzyzewski @joelong01 please review.

File: api/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+mod handlers;
+
+use actix_web::{web, App, HttpServer};
+
+#[actix_rt::main]
+async fn main() -> std::io::Result<()> {
+    std::env::set_var("RUST_LOG", "actix_web=debug");
```
Comment (DanMass) [inline api/src/main.rs:7]: What logging should we expect to see? I'm not seeing things in a CodeSpaces buddy build.

File: game/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+edition = "2018"
+
+[dependencies]
+strum = "0.21"
+strum_macros = "0.21"
+arrayvec="0.7.1"
+rand= "0.8.4"
+serde = { version = "1.0.130", features = ["derive"] }
+serde_json= "1.0.67"
+itertools= "0.10"
```
Comment (DanMass) [inline game/Cargo.toml:13]: \n

Comment (DanMass) [inline game/Cargo.toml:13]: We should replace the last  use of IterTools with the combinator module.

While we're at it, we can clean up combinator in this repo. It's easy to make it clippy clean:
https://github.com/DanMass/fantastic-barnacle/blob/main/src/combinator.rs

File: game/src/lib.rs [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 #![allow(unused_imports)]
 
+#[macro_use]
+mod cards;
+mod combinator;
+mod counting;
+mod scoring;
+mod select_cards;
+
+use cards::Card;
```
Comment (DanMass) [inline game/src/lib.rs:11]: prefer arranging by alpha

**Review by DanMass — APPROVED** [review]

File: api/src/main.rs (hunk 1 above)
Comment (aaron-schnieder) [reply api/src/main.rs:7]: I thought we would get logging in the Output for all HTTP calls, but I am not seeing it either. Will remove for now and we can handle logging all up as separate issues.

**Pull request merged**